Source/ENS/Public/Characters/Player/EnsPlayerController.h

```diff
@@ -104,4 +107,41 @@ class ENS_API AEnsPlayerController : public APlayerController
 	/// \brief The object the player is trying to interact with.
 	UObject* PendingInteractObject = nullptr;
 #pragma endregion
+	
+#pragma region MeshVisibility
+public:
+	/// @brief Speed at which meshes fade in when no longer occluding the camera.
+	UPROPERTY(BlueprintReadWrite, Category="Camera Occlusion|Components")
+	float FadeInSpeed = 5.0f;
+
+	/// @brief Speed at which meshes fade out when occluding the camera.
+	UPROPERTY(BlueprintReadWrite, Category="Camera Occlusion|Components")
+	float FadeOutSpeed = 3.0f;
+
+	/// @brief Minimum opacity value for occluding meshes (0.0 = fully transparent, should be > 0 to remain visible).
+	UPROPERTY(BlueprintReadWrite, Category="Camera Occlusion|Components")
+	float MinOpacity = 0.1f;
+
+	/// @brief Maximum opacity value for non-occluding meshes (1.0 = fully opaque)
+	UPROPERTY(BlueprintReadWrite, Category="Camera Occlusion|Components")
+	float MaxOpacity = 1.f;
+
+private:
+	/// @brief Recursively collects all StaticMeshComponents from an Actor and its children
+	/// @param Actor The root Actor to search from
+	/// @param Result Array to store found StaticMeshComponents
+	/// @return True if any components were found
+	bool GetChildsOfActor(const AActor* Actor, TArray<UStaticMeshComponent*>& Result);
```

Comment:
just return the array and the boolean can be obtained (with a small diff) with the size of the array. Anyway, you do not use it.